packages/@react-aria/grid/src/useGridSelectionCheckbox.ts

```diff
@@ -31,8 +32,30 @@ export function useGridSelectionCheckbox<T, C extends GridCollection<T>>(props:
   let isDisabled = !state.selectionManager.canSelectItem(key);
   let isSelected = state.selectionManager.isSelected(key);
 
+  let isShiftDown = useRef(false);
+  useEffect(() => {
+    let trackKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        isShiftDown.current = true;
+      }
+    };
+    let trackKeyUp = (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        isShiftDown.current = false;
+      }
+    };
+
+    document.addEventListener('keydown', trackKeyDown);
+    document.addEventListener('keyup', trackKeyUp);
+
+    return () => {
+      document.removeEventListener('keydown', trackKeyDown);
+      document.removeEventListener('keyup', trackKeyUp);
+    };
+  });
+
   // Checkbox should always toggle selection, regardless of selectionBehavior.
```

Comment:
Will need to discuss with the team if the proposed change conflicts with this intent

Reply:
Just to add some background here: I have a table with `href` on each row so people use the checkboxes to do the initial selection. While the table then switches into another mode which would allow selection also on the rest of reach row, people tend to keep clicking on the checkboxes. Not having `Shift+Click` available there while it is on the rest of the table is kinda confusing and led to complaints from customers and even the CEO 😅